src/lib/common/sol-types.c

```diff
@@ -174,11 +174,12 @@ sol_drange_equal(const struct sol_drange *var0, const struct sol_drange *var1)
     ((var1 > 0) && (var0 < INT32_MIN + var1))
 
 #define SOL_IRANGE_MULTIPLICATION_OVERFLOW(var0, var1) \
+    ((var0 == INT32_MAX) && (var1 == -1)) || \
```

Comment:
Another thing that should be fixed here while you're at it: either convert these overflow macros to `static inline` functions (which are [as fast as macros](https://gcc.gnu.org/onlinedocs/gcc/Inline.html)), or add parenthesis to these macro arguments.